fosite.go

```diff
@@ -82,6 +82,20 @@ func (a *PushedAuthorizeEndpointHandlers) Append(h PushedAuthorizeEndpointHandle
 	*a = append(*a, h)
 }
 
+// DeviceEndpointHandlers is a list of DeviceEndpointHandler
+type DeviceEndpointHandlers []DeviceEndpointHandler
+
+// Append adds an DeviceEndpointHandlers to this list. Ignores duplicates based on reflect.TypeOf.
+func (a *DeviceEndpointHandlers) Append(h DeviceEndpointHandler) {
+	for _, this := range *a {
+		if reflect.TypeOf(this) == reflect.TypeOf(h) {
```

Comment:
yuck, reflect.. couldn't we extend the DeviceEndpointHandler with a method that would return a type string.. then DeviceEndpointHandlers on L86 would become `map[string]DeviceEndpointHandler` which would be a map of type to DeviceEndpointHandler

Reply:
I am not great at go, but I can understand why you wouldn't want to use reflect. 
But this is the same pattern that is used in all the other handlers `Append` function, I would like to keep it this way for consistency's sake. We would like to merge this with upstream in the future, so IMO we should try to keep any existing patters.

Reply:
using reflect is generally frowned upon and should only be used as last resort..